google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryAggregationsTest.java

```diff
@@ -81,6 +81,25 @@ public void canRunCountUsingAggregationMethod() throws Exception {
     assertThat(snapshot.getCount()).isEqualTo(2);
   }
 
+  @Test
+  public void allowsAliasesLongerThan1500Bytes() throws Exception {
+    // The longest field name allowed is 1500. The alias chosen by the client is <op>_<fieldName>.
+    // If the field name is
+    // 1500 bytes, the alias will be longer than 1500, which is the limit for aliases. This is to
+    // make sure the client
+    // can handle this corner case correctly.
+    String longField = "";
+    for (int i = 0; i < 150; i++) {
+      longField += "0123456789";
+    }
+    Map<String, Map<String, Object>> testDocs = map("a", map(longField, 1), "b", map(longField, 2));
```

Comment:
I expected the field name to have the same 1500-byte limit as specified in document.proto. I'm unsure how this test works.

Reply:
Sorry, that was an incomplete thought from me. According to [this string size calculation](https://firebase.google.com/docs/firestore/storage-size#document-id-size), this 1500 character string should result in a string length of 1501-bytes. This test may fail in production because the field size is too large.

Reply:
Should we took 16 into consideration?
In your link example "6 + 5 + 6 + 11 + 16 = 44"

Reply:
I tested field name limits after posting my last comment. I found that the server did not add to the byte count when calculating the field name size. That is, in production, I was able to create a field name that was 1500-characters (and 1500 utf-8 encoded bytes).

Therefore, this test is good as is.